{{cookiecutter.profile_name}}/OSLayer.py

```diff
@@ -35,3 +44,24 @@ def print(string: str):
     @staticmethod
     def get_uuid4_string() -> str:
         return str(uuid.uuid4())
+
+    @staticmethod
+    def tail(path: str, num_lines: int = 10) -> List[bytes]:
+        if not Path(path).exists():
+            raise FileNotFoundError("{} does not exist.".format(path))
+
+        completed_process = subprocess.Popen(
+            ["tail", "-n", str(num_lines), path],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        lines = completed_process.stdout.readlines()
+        exit_code = completed_process.poll()
```

Comment:
PS: there is already a function `OSLayer.run_process()` responsible for running a command and getting `stdout` and `stderr`. I think we should use it to avoid code duplication, but we also need to update it, as running `subprocess.run` with `shell=True` can be unsafe

Reply:
Yeah, using `wait()` is probably a better option. I was actually using it for a while. I can't remember why I changed it to `poll()`...  
I will update it and push the changes.  
I would like to keep it separate from `run_process` if ok with you. The reason is that *at the moment* we are using a system call to `tail`, but that could change in the future if we find there are problems with that approach and want to shift to using a pure python solution.  
After writing that though, I guess there is no reason why I can't use `run_process`....  
I'll have a play and push whatever I come up with.

Reply:
Ok, both approaches are fine for me! I am actually running `pandora compare` and the `.o` file has some GBs of size. If bjobs fails, then the log checker will certainly fail, or be very, very slow. Loading the whole file into memory could be also the reason why the log checker gets slow... really looking forward to these changes, will start using this version ASAP!

Reply:
Haha yeah, when I was saying I had examples which are GB in size I was thinking of pandora logs. I think I even had one that was approaching TB once 😅 (trace logging)

Reply:
Ok, could you try out that change?